### PR TITLE
Fixing all stylelint errors

### PR DESCRIPTION
**Pull request by NeOMakinG**
#72 needs to be merged first!

QA : Compare index.html on develop branch with this branch's index.html, if there are a lot of differences then tell me (it should probably not be different visually, it's just linting)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   ],
   "scripts": {
     "watch": "webpack -w --progress --debug",
-    "build": "NODE_ENV=production webpack --progress --debug"
+    "build": "NODE_ENV=production webpack --progress --debug",
+    "css-lint": "stylelint \"**/*.scss\" --cache --cache-location .cache/.stylelintcache"
```
Comment (Progi1984) [inline package.json]:
```suggestion
    "scss-lint": "stylelint \"**/*.scss\" --cache --cache-location .cache/.stylelintcache"
```

**Comment by PierreRambaud** [comment]
One more review and merge @Progi1984 maybe?

File: scss/_alerts.scss [hunk 2]
```diff
@@ -66,77 +66,77 @@
...
+    content: "\E001";
+    background-color: theme-color-background("danger");
   }
 
   .close,
   &.expandable-alert .read-more,
   &.expandable-alert .read-more:hover,
   &.expandable-alert .read-more:focus  {
-    color: theme-color('danger');
+    color: theme-color("danger");
```
Comment (jolelievre) [inline scss/_alerts.scss]: Im' not sure to understand the purpose, I can see the the css doesn't change only their order..
But 1) why? 2) what is the order rule? it doesn't seem to be alphabetical 🤔
I guess all those changes were performed through automatic tools?

Comment (NeOMakinG) [reply scss/_alerts.scss]: Mmh the change you notified is just a character swap between double quote and single quote?

File: scss/_buttons.scss [hunk 3]
```diff
@@ -1,23 +1,23 @@
 .btn {
-  border-width: $border-width;
   font-weight: $font-weight-semibold;
-  border-radius: $btn-border-radius;
   white-space: nowrap;
+  border-width: $border-width;
+  @include border-radius($btn-border-radius);
 
-  @include hover-focus {
+  @include hover-focus () {
```
Comment (jolelievre) [inline scss/_buttons.scss]: Ooohh interesting This could interest me but I don't know if it should be done in this PR when using something like `<button type="button"><i class="material-icons>edit</i></button>` There is no hover effect (color change) Wheras with an `<a class="tooltip-link"><i class="material-icons>edit</i></a>` There is one, it would be great to have the same effect on pure buttons (even if it's linked to an additional class to avoid messing with the `btn` class)

We discussed about this @Progi1984 here https://github.com/PrestaShop/PrestaShop/pull/16150/files#r353677243 and I have the same need on a CLDR PR

File: scss/_material-checkbox.scss [hunk 4]
```diff
@@ -23,92 +23,82 @@ $md-checkmark-color: $white;
   }
 
   label {
-    margin-bottom: 0;
     padding-left: $md-checkbox-size + 5px;
+    margin-bottom: 0;
   }
 
-  i.md-checkbox-control {
+  .md-checkbox-control {
```
Comment (jolelievre) [inline scss/_material-checkbox.scss]: Why drop the `i` ?

File: scss/_material-checkbox.scss [hunk 5]
```diff
@@ -23,92 +23,82 @@ $md-checkmark-color: $white;
...
-          $md-checkmark-size: $md-checkbox-size - 2*$md-checkbox-padding;
-
-          transform: rotate(-45deg);
-
-          top: ($md-checkbox-size / 2) - ($md-checkmark-size / 4) - $md-checkbox-size/10;
-          left: $md-checkbox-padding;
-          width: $md-checkmark-size;
-          height: $md-checkmark-size / 2;
+    &:disabled {
+      + .md-checkbox-control {
```
Comment (jolelievre) [inline scss/_material-checkbox.scss]: Again here why drop the `i`

File: scss/_material-checkbox.scss [hunk 6]
```diff
@@ -23,92 +23,82 @@ $md-checkmark-color: $white;
...
-        &:before {
-          cursor: not-allowed;
-        }
-      }
+        transform: rotate(-45deg);
 
-      &:checked {
-        + i.md-checkbox-control:before {
-        }
+        /*! @noflip */ // for RTL compatibility
```
Comment (jolelievre) [inline scss/_material-checkbox.scss]: This comment should be moved with the `border-right-style`

File: scss/_ps-tagger.scss [hunk 7]
```diff
@@ -1,18 +1,19 @@
+/* stylelint-disable selector-class-pattern */
```
Comment (jolelievre) [inline scss/_ps-tagger.scss]: Is this used because camel case names are not accepted by default?

File: scss/_select2.scss [hunk 8]
```diff
@@ -1,8 +1,9 @@
+/* stylelint-disable selector-class-pattern */
```
Comment (jolelievre) [inline scss/_select2.scss]: and I guess here it's it's double `--` which is a problem?

File: scss/_spinners.scss [hunk 9]
```diff
@@ -23,24 +23,11 @@
   }
 }
 
-@-webkit-keyframes rotating {
```
Comment (jolelievre) [inline scss/_spinners.scss]: This is not needed any more? I guess it's present in the compile asset?

File: scss/_tables.scss [hunk 10]
```diff
@@ -4,63 +4,66 @@
...
-      &.column-filters {
-        background: $gray-200;
-        th {
-          border-bottom: none;
-          vertical-align: top;
-          padding: {
-            top: 1rem;
-            bottom: 1rem;
-          }
+    .column-filters {
```
Comment (jolelievre) [inline scss/_tables.scss]: Why drop the `tr` ?

File: scss/_tables.scss [hunk 11]
```diff
@@ -4,63 +4,66 @@
...
 
   td,
   th,
   tr {
     vertical-align: middle;
+  }
 
-    &.form-group {
-      text-align: center;
+  .form-group {
```
Comment (jolelievre) [inline scss/_tables.scss]: And again here it is not in the `tr` block so it matches any tag with this class

File: scss/_type.scss [hunk 12]
```diff
@@ -1,27 +1,24 @@
...
+.tooltip-link {
+  > .material-icons {
+    color: $medium-gray;
+    vertical-align: middle;
 
-      &:hover {
-        color: theme-color('primary');
-      }
+    &:hover {
+      color: theme-color("primary");
```
Comment (jolelievre) [inline scss/_type.scss]: That's the hover effect I mentioned earlier which would be nice for `button` tags

File: scss/_type.scss [hunk 13]
```diff
@@ -1,27 +1,24 @@
-a {
-  &.tooltip-link {
-    > .material-icons {
-      color: $medium-gray;
-      vertical-align: middle;
+.tooltip-link {
```
Comment (jolelievre) [inline scss/_type.scss]: Here we drop the `a`

File: scss/_variables.scss [hunk 14]
```diff
@@ -92,20 +94,20 @@ $theme-colors-background: (
...
+  danger: theme-color("danger"),
+  light: theme-color("light"),
+  dark: theme-color("dark")
 );
 
 
 // Options
 $enable-caret:              true !default;
-$enable-rounded:            false;
+$enable-rounded:            true;
```
Comment (jolelievre) [inline scss/_variables.scss]: This is not related to lint I guess?

File: scss/_custom-forms.scss [hunk 15]
```diff
@@ -23,35 +23,35 @@
...
         @extend .btn-primary:disabled;
       }
     }
   }
   .custom-file-label {
-    &:after {
+    &::after {
+
+      // make the button overlap the border
+      top: $input-border-width * -1;
```
[the remark this answers is not in the document]
Comment (PierreRambaud) [reply scss/_custom-forms.scss]: It's not about border here, it's to scale the top depending on the border.

File: scss/_breadcrumb.scss [hunk 16]
```diff
@@ -1,12 +1,12 @@
 .breadcrumb {
   margin: 0;
-  font-size: 0.75rem;
+  font-size: .75rem;
 
   li {
 
-    & + li:before {
+    & + li::before {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply scss/_breadcrumb.scss]: no, ::before is more semantically correct and :before may not be supported on every browsers

File: scss/_material-checkbox.scss (hunk 4 above)
Comment (PierreRambaud) [reply scss/_material-checkbox.scss]: Because of parent `.md-checkbox` it's not needed and shouldn't target directly a tag, but a class as must as possible.

File: scss/_custom-forms.scss (hunk 15 above)
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply scss/_custom-forms.scss]: That thing was there before this PR, I don't know why it looks like its new, but well the border wont be shown at all if I understand because thats the exact size of the border turned to negative value

**Comment by PierreRambaud** [comment]
@jolelievre Having html tag like `a` or `i` or `tr` is pretty useless when you already target a class.

File: scss/_buttons.scss (hunk 3 above)
Comment (NeOMakinG) [reply scss/_buttons.scss]: Yes It's not linked to that PR at all, create an issue or comment on the Button improvement one :)

File: scss/_material-checkbox.scss (hunk 4 above)
Comment (NeOMakinG) [reply scss/_material-checkbox.scss]: Because Bootstrap doesn't accept selecting a tag element with a class at the same time, they use this rule : https://stylelint.io/user-guide/rules/selector-no-qualifying-type

File: scss/_material-checkbox.scss (hunk 5 above)
Comment (NeOMakinG) [reply scss/_material-checkbox.scss]: https://stylelint.io/user-guide/rules/selector-no-qualifying-type

File: scss/_tables.scss (hunk 10 above)
Comment (NeOMakinG) [reply scss/_tables.scss]: same thing as there : https://stylelint.io/user-guide/rules/selector-no-qualifying-type

File: scss/_tables.scss (hunk 11 above)
Comment (NeOMakinG) [reply scss/_tables.scss]: If we follow the bootstrap guideline we can't select something with a tag and a class/id/props at the same time

See https://stylelint.io/user-guide/rules/selector-no-qualifying-type

File: scss/_type.scss (hunk 12 above)
Comment (NeOMakinG) [reply scss/_type.scss]: We'll think about it in the button improvement issue or another issue

File: scss/_type.scss (hunk 13 above)
Comment (NeOMakinG) [reply scss/_type.scss]: https://stylelint.io/user-guide/rules/selector-no-qualifying-type

File: scss/_variables.scss (hunk 14 above)
Comment (NeOMakinG) [reply scss/_variables.scss]: Not related to lint at all, and it will run in conflict with another PR opened linked to buttons. New UI Kit style is full of border-radius, we need to activate that to enable border-radius and some lines in some files there needed to user mixin border-radius() instead of plain which needed that variable to be true

File: scss/_alerts.scss (hunk 2 above)
Comment (jolelievre) [reply scss/_alerts.scss]: No i meant the order of the miscellaneous css properties, but as you explained to me this is Bootsrap guideline

File: scss/_breadcrumb.scss (hunk 16 above)
[the remark this answers is not in the document]
Comment (jolelievre) [reply scss/_breadcrumb.scss]: Ok, good to know

File: scss/_buttons.scss (hunk 3 above)
Comment (jolelievre) [reply scss/_buttons.scss]: Well actually since the tag have been removed and there are only classes, we can then use `tooltip-link` on a button (even though it is not semanticaly correct, so maybe one day we should add something for the btn class, or an alternative btn-class)

File: scss/_custom-forms.scss (hunk 15 above)
[the remark this answers is not in the document]
Comment (jolelievre) [reply scss/_custom-forms.scss]: Ok I get the purpose (and it's not new indeed it was just moved from the bottom of the file)

File: scss/_material-checkbox.scss (hunk 4 above)
Comment (jolelievre) [reply scss/_material-checkbox.scss]: Ok thanks!

File: scss/_material-checkbox.scss (hunk 6 above)
Comment (jolelievre) [reply scss/_material-checkbox.scss]: Only this comment to move please, it doesn't make sense to leave it here but it's still useful

File: scss/_variables.scss (hunk 14 above)
Comment (jolelievre) [reply scss/_variables.scss]: OK, thanks

**Comment by NeOMakinG** [comment]
Sarah told me to pass it to QA Approved because she doesn't have rights to change it

**Comment by PierreRambaud** [comment]
Thanks @NeOMakinG
